src/util/amount.rs

```diff
@@ -456,27 +458,26 @@ fn fmt_satoshi_in(
     Ok(())
 }
 
-/// Amount
+/// The `Amount` type can be used to express unsigned Bitcoin amounts, it supports arithmetic and
+/// conversion to various denominations.
 ///
-/// The [Amount] type can be used to express Bitcoin amounts that supports
-/// arithmetic and conversion to various denominations.
+/// This type implements several arithmetic operations from [`core::ops`]. To prevent errors due to
+/// overflow when using these operations, it is advised to instead use the checked arithmetic
+/// methods whose names start with `checked_`.
 ///
+/// # Panics
 ///
-/// Warning!
-///
-/// This type implements several arithmetic operations from [core::ops].
-/// To prevent errors due to overflow or underflow when using these operations,
-/// it is advised to instead use the checked arithmetic methods whose names
-/// start with `checked_`.  The operations from [core::ops] that [Amount]
-/// implements will panic when overflow or underflow occurs.  Also note that
-/// since the internal representation of amounts is unsigned, subtracting below
-/// zero is considered an underflow and will cause a panic if you're not using
-/// the checked arithmetic methods.
-///
+/// The operations from [`core::ops`] that `Amount` implements will panic when overflow occurs. Also
+/// note that since the internal representation of `Amount` is unsigned, subtracting below zero is
+/// considered an overflow and will cause a panic if you're not using the checked arithmetic methods.
 #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct Amount(u64);
 
 impl Amount {
+    /// The smallest value that can be represented by `Amount`, `Amount::from_sat(0)`.
+    pub const MIN: Amount = Amount(0);
+    /// The largest value that can be represented by `Amount`, `Amount::from_sat(2.pow(64) − 1)`.
```

Comment:
What about ideas to limit `Amount` to 21M? (I was not a fan, but just asking).

Reply:
Oh I meant to make explicit that this PR does _not_ do that, I'll add it to the description. Thanks.

Reply:
I had a much longer comment on `max_value()` at one stage during development including a test that showed that max was same as `u64::MAX` but I removed it. Perhaps we should add that back in?

Reply:
Since the question is not resolved I propose **not* adding `MAX` nor `max_value`. If we added it people could start rely on it and we couldn't change it then.

Reply:
I agree -- let's leave it for now.

Reply:
`max_value` and `min_value` already exist on `Amount` and `SignedAmount` so we have to take a position.

Reply:
Remove them for a release to stop people from relying on them?

Reply:
I get ya, deprecate the current ones and do not add `MAX`/`MIN` (makes sense since we have 21 million thing in progress too). We have `Amount::ZERO` if needed.

Reply:
It gets more curly, we rely on the ability to create a max (u64::max) amount in `transaction.rs`, according to the comment this is a consensus thing
```rust
// This is used as a "null txout" in consensus signing code.
impl Default for TxOut {
    fn default() -> TxOut {
        TxOut { value: Amount::max_value(), script_pubkey: Script::new() }
    }
}
```
Would we want to expose a const for this or use
```rust
// This is used as a "null txout" in consensus signing code.
impl Default for TxOut {
    fn default() -> TxOut {
        TxOut { value: Amount::from_sat(0xffff_ffff_ffff_ffff), script_pubkey: Script::new() }
    }
}
```

Reply:
And there is a more subtle thing. Our `checked_` functions implicitly leak the internal data used to represent `Amount`, as does `max_value()`. Also, since I copied the rustdoc examples from stdlib there are multiple assertions that use the max value to show how/when the `checked_` functions return `None` (i.e., overflow).

Reply:
Oh, crap! I think we should just change signing code to patch it up. I don't have better ideas that don't involve significant performance hits, code bloat and/or wild `unsafe` tricks.